src/rgw/rgw_acl_swift.cc

```diff
@@ -192,10 +192,23 @@ void RGWAccessControlPolicy_SWIFT::to_str(string& read, string& write)
     ACLGrant& grant = iter->second;
     const uint32_t perm = grant.get_permission().get_permissions();
     rgw_user id;
+    string url_spec;
     if (!grant.get_id(id)) {
-      if (grant.get_group() != ACL_GROUP_ALL_USERS)
-        continue;
-      id = SWIFT_GROUP_ALL_USERS;
+      if (grant.get_group() == ACL_GROUP_ALL_USERS) {
+        id = SWIFT_GROUP_ALL_USERS;
+      } else {
+        url_spec = grant.get_referer();
+        if (url_spec.empty())
+          continue;
+
+        if (perm != 0) {
+          if(url_spec[0] == '.')
+            url_spec = "*" + url_spec;
```

Comment:
[Swift handles this a bit different](https://github.com/openstack/swift/blob/2.12.0/swift/common/middleware/acl.py#L79):

```
$ curl -i "${publicURL}/cont-for-acls" -X POST -H "X-Auth-Token: ${token}" -H "X-Container-Read: .r:*.example.com"
HTTP/1.1 204 No Content
Content-Length: 0
Content-Type: text/html; charset=UTF-8
X-Trans-Id: tx65885b8aa26d4c21bf582-0058878fc9
Date: Tue, 24 Jan 2017 17:32:57 GMT

$ curl -i "${publicURL}/cont-for-acls" -X HEAD -H "X-Auth-Token: ${token}"
HTTP/1.1 204 No Content
Content-Length: 0
X-Container-Object-Count: 0
Accept-Ranges: bytes
X-Timestamp: 1484935150.21775
X-Container-Read: .r:.example.com
X-Container-Bytes-Used: 0
Content-Type: text/plain; charset=utf-8
X-Trans-Id: tx8be1cb3d36544cd385a91-0058878fd0
Date: Tue, 24 Jan 2017 17:33:04 GMT
```